src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -4,1096 +4,686 @@
 static constexpr std::string_view NameKey{ "name" };
-static constexpr std::string_view UpdatesKey{ "updates" };
 static constexpr std::string_view GuidKey{ "guid" };
 
-static constexpr std::string_view DisabledProfileSourcesKey{ "disabledProfileSources" };
-
-static constexpr std::string_view SettingsSchemaFragment{ "\n"
-                                                          R"(    "$schema": "https://aka.ms/terminal-profiles-schema")" };
-
-static constexpr std::string_view jsonExtension{ ".json" };
-static constexpr std::string_view FragmentsSubDirectory{ "\\Fragments" };
+static constexpr std::wstring_view jsonExtension{ L".json" };
+static constexpr std::wstring_view FragmentsSubDirectory{ L"\\Fragments" };
 static constexpr std::wstring_view FragmentsPath{ L"\\Microsoft\\Windows Terminal\\Fragments" };
 
-static constexpr std::string_view AppExtensionHostName{ "com.microsoft.windows.terminal.settings" };
+static constexpr std::wstring_view AppExtensionHostName{ L"com.microsoft.windows.terminal.settings" };
+
+// make sure this matches defaults.json.
+static constexpr winrt::guid DEFAULT_WINDOWS_POWERSHELL_GUID{ 0x61c54bbd, 0xc2c6, 0x5271, { 0x96, 0xe7, 0x00, 0x9a, 0x87, 0xff, 0x44, 0xbf } };
+static constexpr winrt::guid DEFAULT_COMMAND_PROMPT_GUID{ 0x0caa0dad, 0x35be, 0x5f56, { 0xa8, 0xff, 0xaf, 0xce, 0xee, 0xaa, 0x61, 0x01 } };
 
 // Function Description:
 // - Extracting the value from an async task (like talking to the app catalog) when we are on the
 //   UI thread causes C++/WinRT to complain quite loudly (and halt execution!)
 //   This templated function extracts the result from a task with chicanery.
 template<typename TTask>
-static auto _extractValueFromTaskWithoutMainThreadAwait(TTask&& task) -> decltype(task.get())
+static auto extractValueFromTaskWithoutMainThreadAwait(TTask&& task) -> decltype(task.get())
 {
-    using TVal = decltype(task.get());
-    std::optional<TVal> finalVal{};
-    std::condition_variable cv;
-    std::mutex mtx;
+    std::optional<decltype(task.get())> finalVal;
+    til::latch latch{ 1 };
 
-    auto waitOnBackground = [&]() -> winrt::fire_and_forget {
+    const auto _ = [&]() -> winrt::fire_and_forget {
         co_await winrt::resume_background();
-        auto v{ co_await task };
+        finalVal.emplace(co_await task);
+        latch.count_down();
+    }();
 
-        std::unique_lock<std::mutex> lock{ mtx };
-        finalVal.emplace(std::move(v));
-        cv.notify_all();
-    };
-
-    std::unique_lock<std::mutex> lock{ mtx };
-    waitOnBackground();
-    cv.wait(lock, [&]() { return finalVal.has_value(); });
-    return *finalVal;
+    latch.wait();
+    return finalVal.value();
 }
 
-static std::tuple<size_t, size_t> _LineAndColumnFromPosition(const std::string_view string, ptrdiff_t position)
+// Concatenates the two given strings and returns them as a path.
+std::filesystem::path buildPath(const std::wstring_view& lhs, const std::wstring_view& rhs)

```

Comment:
```suggestion
static std::filesystem::path buildPath(const std::wstring_view& lhs, const std::wstring_view& rhs)
```

don't need this implementation detail leaking!